### PR TITLE
Make audit scheduled and manually runnable

### PR DESCRIPTION
**Pull request by spenserblack**
Moves the security audit into a separate workflow so that it can be run
on a schedule, and also dispatched manually.

***

Since a new security issue could be found and reported for an existing dependency even if that dependency doesn't change, I thought it could be helpful to regularly check the audit.
I scheduled it to be run once daily at 23:59 GMT, but it can be set for a different time/frequency :slightly_smiling_face:

### CODE REVIEW

File: .github/workflows/audit.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Security Audit
+on:
+  push:
+    branches: [ master ]
```
Comment (o2sh) [inline .github/workflows/audit.yml:4]:
```suggestion
    branches: [ master ]
    paths: 
      - '**/Cargo.toml'
      - '**/Cargo.lock'
```

No need to trigger the workflow if the commit doesn't affect the dependencies.

File: .github/workflows/audit.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  pull_request:
+    branches: [ master ]
```
Comment (o2sh) [inline .github/workflows/audit.yml:6]:
```suggestion
    branches: [ master ]
    paths: 
      - '**/Cargo.toml'
      - '**/Cargo.lock'
```

No need to trigger the workflow if the commit doesn't affect the dependencies.

File: .github/workflows/audit.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  schedule:
+    - cron: '59 23 * * *'
```
Comment (o2sh) [inline .github/workflows/audit.yml:8]:
```suggestion
    - cron: '0 0 * * 0'
```

What about once a week ?

**Review by o2sh — COMMENTED** [review]
It's indeed safer to have the audit checks be triggered periodically.

I left a few notes.

File: .github/workflows/audit.yml (hunk 3 above)
Comment (spenserblack) [reply .github/workflows/audit.yml:11]: Sounds good to me :+1:

**Comment by spenserblack** [comment]
Thanks for the review! Applied the suggested changes.

**Pull request merged**